tests/services/files/test_file_service.py

```diff
@@ -450,3 +452,108 @@ def test_read_not_committed_external_file(
     # Retrieve file
     with pytest.raises(PermissionDeniedError):
         file_service.get_file_content(identity_simple, recid, "article.txt")
+
+
+def test_deleted_records_file_flow(
+    file_service, location, example_file_record, identity_simple, superuser_identity
+):
+    """Test the lifecycle of a file.
+
+    - Initialize file saving
+    - Save 1 files
+    - Commit the files
+    - List files of the record (ERROR)
+    - Read file metadata (ERROR)
+    - Retrieve a file (ERROR)
+    - Delete a file
+    """
+    # Hack to override the mock service config and set records as deleted
+    current_app.extensions["invenio-records-resources"].registry._services[
+        "mock-records"
+    ].record_cls.deletion_status.is_deleted = True
+    recid = example_file_record["id"]
+    file_to_initialise = [
+        {
+            "key": "article.txt",
+            "checksum": "md5:c785060c866796cc2a1708c997154c8e",
+            "size": 17,  # 2kB
+            "metadata": {
+                "description": "Published article PDF.",
+            },
+        }
+    ]
+    # Initialize file saving
+    result = file_service.init_files(identity_simple, recid, file_to_initialise)
+    assert result.to_dict()["entries"][0]["key"] == file_to_initialise[0]["key"]
+    # for to_file in to_files:
+    content = BytesIO(b"test file content")
+    result = file_service.set_file_content(
+        identity_simple,
```

Comment:
normally I would say you should not be able to set_file_content on deleted record unless for some reason the admin's job requires it. I would ask the zenodo support team for their use cases
I see two options
1. this endpoint should be protected by permissions, and only admin should be able to set content/commit - in this case we miss a test to check if normal user or owner of the record still can commit files
2. nobody, even the admin/moderator (except superuse) should be able to commit or set content

Reply:
This feels a bit out of scope for this [issue](https://github.com/inveniosoftware/invenio-rdm-records/issues/1479), the goal is to hide the files of deleted records. The behaviour of who can update files of deleted records was not changed, at the moment it's only admins. 

If we want to change this we should add the tests in `invenio-rdm-records` as here we mocking the permissions.